app/Services/DomainService.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace App\Services;
+
+use Carbon\Carbon;
+use Iodev\Whois\Factory;
+use App\Models\Monitor;
+use App\Notifications\Notifiable;
+use App\Notifications\DomainExpiration;
+
+class DomainService
+{
+    protected $whois;
+
+    public function __construct()
+    {
+        $this->whois = Factory::get()->createWhois();
+    }
+
+    public function lookupDomain($url)
+    {
+        $baseDomain = $this->getBaseDomainFromUrl($url);
+
+        $domainInfo = $this->whois->loadDomainInfo($baseDomain);
+
+        if ($domainInfo && $domainInfo->expirationDate) {
+            return response()->json([
```

Comment:
Yes, I could have used an array instead of response object. I misinterpreted that I'm sending a request to a different package, so I should get the result as a response object. 
But now I know that the response is used when there is an HTTP request, and so to send the result of the request, we use the response object.